### DISCUSSION TITLE
release assets format unclear

### DISCUSSION THREAD

**Discussion by jo746** [python-gitlab/python-gitlab#2720] [Q&A]
I've been trying to create a release and upload assets using both the CLI and the library. The assets argument is very unclear and I can't seem to make it work.

I always get a `Impossible to create object (400: assets is invalid)`

`gitlab project-release create --project-id <id> --tag-name <tag> --name <release name> --description <description> --assets <what goes here>`

same goes for the library `assets` argument. I cannot figure out the format for the files that I want to include.

**Comment by nejch** [comment]
@jo746 with the CLI unfortunately these kinds of complex arguments are not supported at the moment.

Using the API you should be able to just use a dictionary that resembles the JSON structure expected in https://docs.gitlab.com/ee/api/releases/#create-a-release, but even the upstream docs are not very extensive on this. You can check the curl example to get an idea probably. What have you tried using the library approach?

**Reply by jo746** [reply]
I had just tried passing a list and a simple dict; though I was unsure what the dict needed to look like.  I will try using the format from the rest api docs.  I was hoping that it might support directly uploading files rather than just links though; similar to the https://gitlab.com/gitlab-org/cli tool.
